src/libpgmoneta/utils.c

```diff
@@ -3003,6 +3004,128 @@ pgmoneta_is_file_archive(char* file_path)
    return false;
 }
 
+/* Parser for pgmoneta-cli amd pgmoneta-admin commands */
+bool
+parse_command(int argc,
+              char** argv,
+              int offset,
+              struct pgmoneta_parsed_command* parsed,
+              const struct pgmoneta_command command_table[],
+              size_t command_count)
+{
+#define EMPTY_STR(_s) (_s[0] == 0)
+
+   char* command = NULL;
+   char* subcommand = NULL;
+   bool command_match = false;
+   int default_command_match = -1;
+   int arg_count = -1;
+   int command_index = -1;
+
+   /* Parse command, and exit if there is no match */
+   if (offset < argc)
+   {
+      command = argv[offset++];
+   }
+   else
+   {
+      warnx("A command is required\n");
+      return false;
+   }
+
+   if (offset < argc)
+   {
+      subcommand = argv[offset];
+   }
+
+   for (int i = 0; i < command_count; i++)
+   {
+      if (strncmp(command, command_table[i].command, MISC_LENGTH) == 0)
+      {
+         command_match = true;
+         if (subcommand && strncmp(subcommand, command_table[i].subcommand, MISC_LENGTH) == 0)
+         {
+            offset++;
+            command_index = i;
+            break;
+         }
+         else if (EMPTY_STR(command_table[i].subcommand))
+         {
+            /* Default command does not require a subcommand, might be followed by an argument */
+            default_command_match = i;
+         }
+      }
+   }
+
+   if (command_match == false)
+   {
+      warnx("Unknown command '%s'\n", command);
+      return false;
+   }
+
+   if (command_index == -1 && default_command_match >= 0)
+   {
+      command_index = default_command_match;
+      subcommand = "";
+   }
+   else if (command_index == -1)  /* Command was matched, but subcommand was not */
+   {
+      if (subcommand)
+      {
+         warnx("Unknown subcommand '%s' for command '%s'\n", subcommand, command);
+      }
+      else  /* User did not type a subcommand */
+      {
+         warnx("Command '%s' requires a subcommand\n", command);
+      }
+      return false;
+   }
+
+   parsed->cmd = &command_table[command_index];
+
+   /* Iterate until find an accepted_arg_count that is equal or greater than the typed command arg_count */
+   arg_count = argc - offset;
+   int j;
+   for (j = 0; j < MISC_LENGTH; j++)
```

Comment:
Define `int j` in the `for` loop

Reply:
Or move it to the top of the function

Reply:
Got it. I'll move j to the top since it is used later.